### ISSUE TITLE
Plugin name lowercasing seems to break under non-English locales

### ISSUE THREAD

**Issue by CarrotOfWisdom** [loot/loot#1104]
It was working just a few days ago. All of a sudden LOOT stopped working correctly
I get

Error
Oh no, something went wrong! This is usually because "C:\Users\Username\AppData\Local\Skyrim Special Edition\plugins.txt" is set to be read-only. If it is, unset it and try again. If it isn't, you can check your LOOTDebugLog.txt (you can get to it through the main menu) for more information.

everytime I try to apply the load order.

DebugLog:
[LOOTDebugLog.txt](https://github.com/loot/skyrimse/files/3170086/LOOTDebugLog.txt)

Things I tried:

-Looking at the said folder and find out that there is no plugins.txt

-Uncheck the "read-only" on the folder

-Create a plugins.txt

-Run LOOT as admin

-Try to use the "sort" in Mod Organizer 2 (gives "Error code: 1")

-Deleting Skyrim Special Edition folder in Appdata/Local

-Reinstalling LOOT again and again

-Deleting Plugins.txt and Loadorder.txt in MO2 profile

-Manually writing the load order in plugins.txt and loadorder.txt

My Load Order:

[load order.txt](https://github.com/loot/skyrimse/files/3170091/load.order.txt)

**Comment by CarrotOfWisdom** [comment]
> What version of LOOT is this with?

Version 0.14.4 (build ec99692e)

**Comment by Ortham** [comment]
For some reason it's trying to set

```
[16:09:56.919191] [info]: 		ccmtysse002-ve.esl
[16:09:56.919199] [info]: 		ccffbsse001-imperialdragon.esl
```

When Skyrim.ccc specifies

```
ccFFBSSE001-ImperialDragon.esl
ccMTYSSE002-VE.esl
```

Can you upload your full debug log, including what gets logged when you sort your load order?

**Comment by CarrotOfWisdom** [comment]
> For some reason it's trying to set
> 
> ```
> [16:09:56.919191] [info]: 		ccmtysse002-ve.esl
> [16:09:56.919199] [info]: 		ccffbsse001-imperialdragon.esl
> ```
> 
> When Skyrim.ccc specifies
> 
> ```
> ccFFBSSE001-ImperialDragon.esl
> ccMTYSSE002-VE.esl
> ```
> 
> Can you upload your full debug log, including what gets logged when you sort your load order?

[LOOTDebugLog.txt](https://github.com/loot/loot/files/3170130/LOOTDebugLog.txt)

**Comment by Ortham** [comment]
```
[16:38:18.641818] [trace]: Skipping adding hardcoded plugin edges for "ccFFBSSE001-ImperialDragon.esl" as it is not installed.
```

I'm not sure why it's thinking `ccffbsse001-imperialdragon.esl` isn't installed there when other parts have it right... It's also the only plugin that it's gotten wrong.

Try replacing your `loot.dll` with [this one](https://bintray.com/loot/snapshots/download_file?file_path=libloot-0.14.6-1-ga36f6b7_fix-hardcoded-plugin-check-win32.7z) then sort and upload your log again, I've added extra detail to the logged message.

**Comment by CarrotOfWisdom** [comment]
[LOOTDebugLog.txt](https://github.com/loot/loot/files/3170425/LOOTDebugLog.txt)

**Comment by CarrotOfWisdom** [comment]
> ```
> [16:38:18.641818] [trace]: Skipping adding hardcoded plugin edges for "ccFFBSSE001-ImperialDragon.esl" as it is not installed.
> ```
> 
> I'm not sure why it's thinking `ccffbsse001-imperialdragon.esl` isn't installed there when other parts have it right... It's also the only plugin that it's gotten wrong.
> 
>

It seems like it only happens with imperialdragon.esl (CC - Civil War Champions). When I disable it, I can apply the load order, if I enable it I can't.

**Comment by Ortham** [comment]
I've tried replicating this several times, and haven't been able to. I can't tell why the issue would be happening by looking at the code either, so I don't know what more I could do.

The only thing I can think of is that there's something special about `ccffbsse001-imperialdragon.esl` that LOOT's not accounting for, but I can't check that because I don't own whatever Creation Club content that's from.

**Comment by CarrotOfWisdom** [comment]
>
> 
> I don't own whatever Creation Club content that's from.

I can send you 750 CC Credits via Steam to help you fix the issue.

**Comment by Ortham** [comment]
Just in case this works: try renaming `ccffbsse001-imperialdragon.esl` to `ccFFBSSE001-ImperialDragon.esl`.

**Comment by CarrotOfWisdom** [comment]
> Just in case this works: try renaming `ccffbsse001-imperialdragon.esl` to `ccFFBSSE001-ImperialDragon.esl`.

It worked. Thank you so much.

**Comment by Ortham** [comment]
I have no idea why, but I'm glad it did.

Unless... Are you running Windows in a non-English locale/language, and if so, which one?

**Comment by CarrotOfWisdom** [comment]
> I have no idea why, but I'm glad it did.
> 
> Unless... Are you running Windows in a non-English locale/language, and if so, which one?

I'm running Windows in Turkish.

**Comment by Ortham** [comment]
Ah, this'll probably be because `I` doesn't lowercase to `i` in Turkish, so when LOOT tries to do a case-insensitive comparison to find the loaded plugin from its filename, it can't find a match.

This will need fixes in LOOT and libloot - there are a few other cases where strings are lowercased for path comparison, so other things may also not be working correctly.

**Comment by Ortham** [comment]
@CarrotOfWisdom Can you try renaming `ccFFBSSE001-ImperialDragon.esl` back to `ccffbsse001-imperialdragon.esl` and sorting with [this build](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.14.4-7-g6835223_fix-lowercasing.7z)? It should have a proper fix for the issue.

**Comment by CarrotOfWisdom** [comment]
> @CarrotOfWisdom Can you try renaming `ccFFBSSE001-ImperialDragon.esl` back to `ccffbsse001-imperialdragon.esl` and sorting with [this build](https://bintray.com/loot/snapshots/download_file?file_path=loot_0.14.4-7-g6835223_fix-lowercasing.7z)? It should have a proper fix for the issue.

Well, I haven't tried with the build you sent yet but I tried with the same build I had when the problem occured just to make sure if it still happens. Now it works fine for some reason after renaming it back to lowercase. Should I still try the build you sent?

**Comment by Ortham** [comment]
Yes, it's still worth a try.

**Comment by pStyl3** [comment]
Closing this issue, since commit https://github.com/loot/loot/commit/c3967908de6193b3139cec56fe0f3b6ad8d3b300 has been merged and LOOT 0.14.5 incorporates the fix, as well as we haven't heard from @CarrotOfWisdom in a long time.

If the issue should return while using LOOT 0.14.5 or later, please open a new issue.
